### PR TITLE
Apply props in alphabetic order, new offline mode build, 8.5.x folder

### LINKED ISSUE #127: Properties creation on Build and Runtime
Hi,

I have two suggestions about config-properties.

- Runtime: some properties have dependencies and should be applied in a certain order (ex. JDBCProvider => JAASAuthData => DataSource => ConnectionPool). Right now, the applyConfigs function in start_server.sh applies config-properties files in semi-random order. I'd suggest it applies in alphabetic order, so we can use numbers to define de correct order.

```
applyConfigs(){
  if [ ! -z "$(ls /etc/websphere)" ]; then
    echo "+ Found config-files under /etc/websphere. Executing..."
    for i in $(ls /etc/websphere/*props); do
    /work/applyConfig.sh $i
    done
  fi
}
```
This way, we could name the files like below and be sure there'll be no error (example, creating a connection pool before creating JDBC Provider, AuthData and DataSource)
```
01-JDBCProvider.props
02-JAASAuthData.conf
03-DataSource.conf
04-ConnectionPool.conf
```

### LINKED ISSUE #128: Decouple installation download and image build
Hi,

I really liked when tWAS download through IM was decoupled from the image build.

Right now, in my company's private repo, we have a separate Installation Manager image to download and compress tWAS to a local folder, which we can later move to an artifact folder during our CI.

The Dockerfile looks like this:

```
# build image to extract the tar ball.
FROM ubuntu:16.04 as builder
 ARG WAS_VERSION=9.0.0.9
 ARG WAS_TAR_BALL=was-${WAS_VERSION}.tar.gz
COPY was/${WAS_TAR_BALL} /
RUN tar -xzf "$WAS_TAR_BALL"
# patch wsadmin.sh to avoid error when deploying apps
RUN sed -i 's/-Xms256m/-Xms1024m/g' /opt/IBM/WebSphere/AppServer/bin/wsadmin.sh \
    && sed -i 's/-Xmx256m/-Xmx1024m/g' /opt/IBM/WebSphere/AppServer/bin/wsadmin.sh


# final image that will copy extracted tar ball from builder
FROM ubuntu:16.04
ARG USER=was
ARG GROUP=was

RUN apt-get update && apt-get install -y openssl wget unzip

COPY scripts/ /work/

RUN groupadd $GROUP --gid 510 \
  && useradd $USER -g $GROUP -m --uid 500 \
  && mkdir /opt/IBM \
  && chmod -R +x /work/* \
  && mkdir /etc/websphere \
  && mkdir /work/config \
  && chown -R $USER:$GROUP /work /opt/IBM /etc/websphere

COPY --chown=500:510 --from=builder /opt/IBM /opt/IBM

....
```

I can submit a PR, if it's ok.

### PR DESCRIPTION
**Pull request by DSTOLF**
- start_server.sh, applyConfig.sh - will apply props in alphabetic order. closes #127 
- New installation manager to download and keep WAS in local storage;
- New Dockerfile.offline for offline build, using previously downloaded WAS. closes #128  
- Change README to reflect all changes.

### CODE REVIEW

**Comment by arthurdm** [comment]
@arturdzm - can you please review this PR?

File: docker-build/9.0.0.10/Dockerfile.offline [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ARG SERVER_NAME=server1
+ARG ADMIN_USER_NAME=wsadmin
+
+COPY scripts/ /work/
+COPY licenses /licenses/
+
+
+RUN apt-get update && apt-get install -y openssl
+
+COPY scripts/ /work/
```
Comment (arthurdm) [inline docker-build/9.0.0.10/Dockerfile.offline]: is this a dup of line 25?

Comment (DSTOLF) [reply docker-build/9.0.0.10/Dockerfile.offline]: Yes, it was. Just corrected that on a new commit.

**Comment by DSTOLF** [comment]
I think the PR is ready for review again. If it's too many changes for on PR, please let me know and I'll break it down a bit. Here's a list of the changes in `docker-build` folder:
- `9.0.0.10` and `9.0.0.x` - Instalation Manager scripts to download and store WAS binaries, Dockerfile.offline for offline build, multi-stage Dockerfile. Closes #128 ;
- `start_server.sh`, `applyConfig.sh `- will apply props in alphabetic order. closes #127
- `8.5.x` - for users interested in using older versions of WAS to build the image themselves. Closes #141

**Comment by arthurdm** [comment]
ah, interesting - I wasn't aware of that limitation of `COPY`...thx for pointing that out!  I have now added a `+1` to the [GH issue](https://github.com/moby/moby/issues/35018).

Until docker is fixed, if you could change it to be `1001:0`, that would be great.  

@arturdzm -after the uID/gID changes, this PR looks good to me.    Can you please verify?

**Comment by DSTOLF** [comment]
@arthurdm 

Just to be sure, it's uid=1001 and gid=0, right?

If so, all the Dockerfiles that I have changed to multi-stage build are fixed now.

9.0.0.9/Dockerfile and 8.5.x/Dockerfile (which was based on 9.0.0.9) are not multi-stage, so I didn't see any reason fiddle with uid and groupid there.

Regards!

**Comment by arturdzm** [comment]
@DSTOLF  
Thank you for the update, yes uid,gid are correct now.

The rest looks good @arthurdm

**Comment by arthurdm** [comment]
awesome.   thanks so much @DSTOLF!

**Pull request merged**